### PR TITLE
Add support for prefixing attribute in ErrorSource.from_attribute

### PR DESCRIPTION
**Pull request by katafrakt**
This is required to have support for nested errors in changesets in Surgex.Changeset.

See https://github.com/surgeventures/surgex/pull/29

### CODE REVIEW

File: lib/jabbax/document/error_source.ex [hunk 1]
```diff
@@ -1,7 +1,13 @@
 defmodule Jabbax.Document.ErrorSource do
   defstruct [:pointer, :parameter]
 
-  def from_attribute(attribute) do
-    %__MODULE__{pointer: "/data/attributes/#{attribute}"}
+  def from_attribute(attribute, prefix \\ []) do
```
Comment (kamilkowalski) [inline lib/jabbax/document/error_source.ex:4]: I'm wondering whether it wouldn't make more sense for this function to accept two types of arguments - a list and an atom/string. The "prefix" here is really a path to the attribute, not a namespace or modifier... We could then have:


```elixir
def from_attribute(path) when is_list(path) do
  pointer = "/data/attributes/#{Enum.join(path, "/")}"
  %__MODULE__{pointer: pointer}
end

def from_attribute(attribute), do: from_attribute([attribute])
```

And use it like:

```elixir
ErrorSource.from_attribute(:name) # => %ErrorSource{pointer: "/data/attributes/name"}
ErrorSource.from_attribute([:address, :city]) # => %ErrorSource{pointer: "/data/attributes/address/city"}
```

WDYT?

Comment (katafrakt) [reply lib/jabbax/document/error_source.ex]: Ok, I changed it. But now I think it's a bit weird to pass a list to `from_attribute` function. Perhaps the one accepting a list should be named `from_attributes`?

Comment (kamilkowalski) [reply lib/jabbax/document/error_source.ex]: The trick is to think of it as an attribute _path_, which is a list. If you call the argument `attribute_path` then it makes sense I think - while the old implementation could use the name `attribute` for the argument.

File: lib/jabbax/document/error_source.ex [hunk 2]
```diff
@@ -3,7 +3,16 @@ defmodule Jabbax.Document.ErrorSource do
 
   defstruct [:pointer, :parameter]
 
+  def from_attribute(attributes) when is_list(attributes) do
```
Comment (kamilkowalski) [inline lib/jabbax/document/error_source.ex:6]: This is not a list of attributes, it's a path to the attribute in question:

```suggestion
  def from_attribute(attribute_path) when is_list(attribute_path) do
```

**Review by kamilkowalski — CHANGES_REQUESTED** [review]

**Review by kamilkowalski — APPROVED** [review]

**Review by kamilkowalski — CHANGES_REQUESTED** [review]
You'll need to bump the minor version for it to be released automatically.

**Review by kamilkowalski — APPROVED** [review]
🙌

**Pull request merged**